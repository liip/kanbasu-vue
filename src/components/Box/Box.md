```vue
<box variant="default">
  Hello
</box>
```

```vue
<box variant="primary" size="small">
  Hello
</box>
```