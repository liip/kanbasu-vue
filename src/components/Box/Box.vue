<template>
  <div :class="classNames">
    <slot></slot>
  </div>
</template>

<script>
export default {
  name: 'Box',

  props: {
    /**
     * The style of the box:
     * `default, primary`
     */
    variant: {
      type: String,
    },
    /**
     * The box padding:
     * `tight, tiny, small, large, huge`
     */
    size: {
      type: String,
    },
  },

  computed: {
    classNames() {
      return [
        'box',
        { [`box--${this.variant}`]: !!this.variant },
        { [`box--${this.size}`]: !!this.size },
      ];
    },
  },
};
</script>
