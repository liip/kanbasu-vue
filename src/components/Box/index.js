import Box from './Box.vue';

Box.install = function(Vue) {
  Vue.component(Box.name, Box);
};

export default Box;
