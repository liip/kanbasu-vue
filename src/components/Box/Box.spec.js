import { shallowMount } from '@vue/test-utils';
import Box from './Box.vue';

describe('Box.vue', () => {
  it('renders', () => {
    const wrapper = shallowMount(Box);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with variant', () => {
    const wrapper = shallowMount(Box, {
      propsData: {
        variant: 'primary',
      },
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with size', () => {
    const wrapper = shallowMount(Box, {
      propsData: {
        size: 'small',
      },
    });
    expect(wrapper).toMatchSnapshot();
  });
});
