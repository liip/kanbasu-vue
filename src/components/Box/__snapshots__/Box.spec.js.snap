// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box.vue renders 1`] = `<div class="box"></div>`;

exports[`Box.vue renders with size 1`] = `<div class="box box--small"></div>`;

exports[`Box.vue renders with variant 1`] = `<div class="box box--primary"></div>`;
